components/omega/doc/design/VerticalMixingCoeff.md

```diff
@@ -0,0 +1,182 @@
+
+### 2.3 Requirement: Vertical mixing models cannot ingest single columns at a time
+
+Many standard vertical mixing libraries (e.g. CVMix) receive and operate on one column at a time. This results in poor performance, especially on accelerators. Any utilized model of vertical mixing must ingest and operate on multiple columns concurrently.
+
+## 3 Algorithmic Formulation
+
+For Omega-1 a few simple vertical mixing algorithms will be used.  
+
+### 3.1 Richardson number dependent mixing
+
+One of the simplest class of models of vertical turbulence are polynomial functions of the gradient Richardson number. As in MPAS-Ocean, we choose to define the Richardson number as
+
+$$
+Ri = \frac{N^2}{\left|\frac{\partial \mathbf{U}}{\partial z}\right|^2}
+$$
+
+where $N^2$ is the Brunt Vaisala Frequency, which for non Boussinesq flows is defined as
+
+$$
+N^2 = \frac{g}{\rho_0}\frac{\partial \rho}{\partial z}
+$$
+
+This term is discretized as
+
+$$
+N^2(k) = g \rho_0 \frac{\rho_{DD}(k) - \rho(k)}{z_m(k-1) - z_m(k)}
+$$
+
+Here $\rho_{DD}$ is the density of the fluid of layer k-1 displaced to layer k adiabatically.
+
+The shear in the denominator is discretized as
+
+$$
+\left| \frac{\partial \mathbf{U}}{\partial z}\right|^2 = \left(\frac{U_n(k-1)-U_n(k)}{z_m(k-1) - z_m(k)}\right)^2 + \left(\frac{U_t(k-1)-U_t(k)}{z_m(k-1) - z_m(k)}\right)^2
+$$
+
+Where the subscripts *n* and *t* are the normal and tangential velocities respectively.
+
+With the definition of the gradient richardson number, the viscosity and diffusivity are defined according to [Pacanowski and Philander (1981)](https://journals.ametsoc.org/view/journals/phoc/11/11/1520-0485_1981_011_1443_povmin_2_0_co_2.xml?tab_body=pdf) as
+
+$$
+\nu = \frac{\nu_o}{(1+\alpha Ri)^n} + \nu_b
+$$
+
+$$
+\kappa = \frac{\nu}{(1+\alpha Ri)} + \kappa_b
+$$
+
+in these formulae, $\alpha$ and *n* are tunable parameters, most often chosen as 5 and 2 respectively. $\nu_b$ and $\kappa_b$ are the background viscosity and diffusivity respectively. MPAS-Ocean has a $\kappa_{b,passive}$ for applying to just the passive tracers, leaving the active tracers with $\kappa_b$.
+
+### 3.2 Convective Instability Mixing
+
+Commonly, mixing due to convective instability is treated as a step function for the diffusivity and viscosity. This is often represented as
+
+$$
+\kappa =
+\begin{cases}
+\kappa_{conv} \quad \text{ if } N^2 \leq N^2_{crit}\\
+0 \quad \text{ if } N^2 > N^2_{crit}
+\end{cases}
+$$
+
+A similar expression is utilized for viscosity. The effect of this formula is to homogenize T, S, passive tracers, and normal velocity, for unstable stratification, but also in neutral stratification. The behavior in unstable stratification is likely correct, but for neutral stratification, homogenization of momentum is questionable. To allow for a different behavior, we slightly modifty the algorithm above to,
```

Comment:
"different behavior" from MPAS-Ocean or the literature?

Reply:
I believe this is different behavior from the literature/other models... but this was written by @vanroekel, so I'll let him clarify

Reply:
this is different from the standard assumption in things like CVMix where you turn on convective mixing even for neutral stratification, here I propose we only turn it on when less than zero, even slightly.  Not sure how best to phase this.  But as to your question @cbegeman I guess it is yes to both.  MPAS uses zero as the cutoff, there's not a lot of literature on this, but the CVMix source does have an option for a non zero criterion for convective mixing so maybe not too different

Reply:
I see no need to follow CVMix exactly.  I'd go with (2)

Reply:
For context, I believe (if I remember correctly) I did a test run with option 1 in MPAS-O a while back and it made the diffusivity profiles much less noisy (in a good way), though it did not appear to have much of an effect beyond that. So this change (done either way) does have some evidence to support it in MPAS-O.

Reply:
> I see no need to follow CVMix exactly. I'd go with (2)

Ok, I will add some wording in to clarify this, unless others object to this option.